juju/model.py

```diff
@@ -1982,19 +1982,29 @@ async def _deploy(self, charm_url, application, series, config,
             raise JujuError('\n'.join(errors))
         return await self._wait_for_new('application', application)
 
-    async def destroy_unit(self, *unit_names):
+    async def destroy_unit(self, unit_id, destroy_storage=False, dry_run=False, force=False, max_wait=None):
         """Destroy units by name.
 
         """
         connection = self.connection()
         app_facade = client.ApplicationFacade.from_connection(connection)
 
-        log.debug(
-            'Destroying unit%s %s',
-            's' if len(unit_names) == 1 else '',
-            ' '.join(unit_names))
+        # Get the corresponding unit tag
+        unit_tag = utils.unit_id_to_unit_tag(unit_id)
+        if unit_tag is None:
+            log.error("Error converting %s to a valid unit tag", unit_id)
+            return JujuUnitError("Error converting %s to a valid unit tag", unit_id)
 
-        return await app_facade.DestroyUnits(unit_names=list(unit_names))
+        log.debug(
+            'Destroying unit %s', unit_id)
+
+        return await app_facade.DestroyUnit(units=[{
+            'unit-tag': unit_tag,
+            'destroy-storage': destroy_storage,
+            'force': force,
+            'max-wait': max_wait,
+            'dry-run': dry_run,
+        }])
```

Comment:
The tests that are using this code in `test_model.py` (through `application.destroy_unit`), namely the `test_wait_for_idle_with_exact_units_scale_down` and the `test_wait_for_idle_with_exact_units_scale_down_zero` need to be updated. We probably should have at least 1 test just for the `model.destroy_unit` itself.

Reply:
We should probably address this in a different PR.